cmd/helm/reference.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright The Helm Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/Masterminds/semver"
+	"github.com/gosuri/uitable"
+
+	"k8s.io/helm/pkg/chart"
+	"k8s.io/helm/pkg/chart/loader"
+)
+
+type refListOptions struct {
+	chartpath string
+}
+
+func (o *refListOptions) run(out io.Writer, isLib bool) error {
+	c, err := loader.Load(o.chartpath)
+	if err != nil {
+		return err
+	}
+
+	var reqs []*chart.Dependency
+	var dirName string
+	if isLib {
+		reqs = c.Metadata.Libraries
+		dirName = "library"
+	} else {
+		reqs = c.Metadata.Dependencies
+		dirName = "charts"
+	}
+
+	if reqs == nil {
+		if isLib {
+			fmt.Fprintf(out, "WARNING: no libraries at %s\n", filepath.Join(o.chartpath, dirName))
+		} else {
+			fmt.Fprintf(out, "WARNING: no dependencies at %s\n", filepath.Join(o.chartpath, dirName))
+		}
+		return nil
+	}
+
+	o.printDependencies(out, reqs, dirName)
+	fmt.Fprintln(out)
+	o.printMissing(out, reqs, dirName)
+	return nil
+}
+
+func (o *refListOptions) dependencyStatus(dep *chart.Dependency, dirName string) string {
+	filename := fmt.Sprintf("%s-%s.tgz", dep.Name, "*")
+	archives, err := filepath.Glob(filepath.Join(o.chartpath, dirName, filename))
+	if err != nil {
+		return "bad pattern"
+	} else if len(archives) > 1 {
+		return "too many matches"
+	} else if len(archives) == 1 {
```

Comment:
There's a lot of return statements in conditional code here. This could be cleaned up quite a bit by removing the `else ifs`. Something like this:

```
if err != nil {
    return "bad pattern"
}
if len(archives) > 1 {
    return "too many matches"
}
// etc...
```